Automattic/jetpack
# Sandbox: create a new sandbox package

#### Changes proposed in this Pull Request:
 * Move the Jetpack sandbox functions to a new package. This will developers to use sandbox servers when working with monorepo plugins that use the connection.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
With this PR, we should be able to use a sandbox server with any of the connected plugins in the monorepo. So, testing will include a few different plugins.

##### Test with Jetpack Backup
1. Install and activate this branch of the Jetpack Backup plugin.
2. Set the `JETPACK__SANDBOX_DOMAIN` constant to your sandbox domain.
3. Set up your sandbox.
4. In wp-admin, navigate to the Jetpack Backup menu and connect to WPCOM.
5. Check your site's `debug.log` file for log entries that indicate the request was sandboxed.
6. Verify that the top admin menu shows the `Jetpack API Sandboxed` message. Mouse over the message and verify that the tooltip shows your sandbox url.
7. Deactivate the Jetpack Backup plugin.

##### Test with Jetpack
8. Install and activate this branch of the Jetpack plugin.
9. In wp-admin, navigate to the Jetpack menu. Disconnect from WPCOM and reconnect.
10. Check your site's `debug.log` file for log entries that indicate the request was sandboxed.
11. Verify that the top admin menu shows the `Jetpack API Sandboxed` message. Mouse over the message and verify that the tooltip shows your sandbox url.
